packages/react/src/components/hooks/useWebRTC.ts

```diff
@@ -0,0 +1,62 @@
+import { useEffect, useState } from "react";
+import { WebRTC } from "@signalwire/js";
+
+export default function useWebRTC(
+  deviceEnabled: {
+    cameras?: boolean;
+    microphones?: boolean;
+    speakers?: boolean;
+  } = { cameras: true, microphones: true, speakers: true }
+) {
+  const [cameras, setCameras] = useState<MediaDeviceInfo[]>([]);
+  const [microphones, setMicrophones] = useState<MediaDeviceInfo[]>([]);
+  const [speakers, setSpeakers] = useState<MediaDeviceInfo[]>([]);
+
+  useEffect(() => {
+    async function setupList() {
+      deviceEnabled?.cameras !== false &&
+        WebRTC.getCameraDevices().then((c) => setCameras(Array.from(c)));
+      deviceEnabled?.microphones !== false &&
+        WebRTC.getMicrophoneDevices().then((m) =>
+          setMicrophones(Array.from(m))
+        );
+      deviceEnabled?.speakers !== false &&
+        WebRTC.getSpeakerDevices().then((s) => setSpeakers(Array.from(s)));
+    }
+    setupList();
+  }, []);
+
+  // TODO: not in this hook in particular, but there should
+  // be a way to keep track of the devices currently being used.
+
+  useEffect(() => {
+    let deviceWatcher: any;
+    async function setupWatchers() {
+      if (deviceEnabled.cameras !== false) {
```

Comment:
should this `if` check if at least one in `cameras`/`microphones`/`speakers` is not false?